src/App.tsx

```diff
@@ -15,36 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = 'default',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType | null,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField !== SortType.Default) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return good1.localeCompare(good2);
+
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | null>(null);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, isReversed);
+
+  function handleReset(): void {
```

Comment:
The function's name `handleReset` should start with a verb that more accurately describes the action performed by the function. Consider renaming it to `resetSortAndOrder` or similar to clarify its purpose.